### PR TITLE
[ENH] Visual settings: saving settings, custom dialogs

### PR DESCRIPTION
**Pull request by VesnaT**
##### Issue

User is forced to use the predefined dialog when clicking upon the Visual Settings icon in the status bar.

##### Description of changes
- Enable using custom Visual settings dialog
- Enable using saved settings
- Remove OrderedDict

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by markotoplak** [comment]
Thanks, @VesnaT, I think this PR makes it easier to understand where are the settings specified in the widgets. It also allows the dialog to be fully customizable.

@janezd, merge freely if you agree.

File: orangewidget/utils/visual_settings_dlg.py [hunk 1]
```diff
@@ -110,13 +112,40 @@ def show_dlg(self):
         self.activateWindow()
 
 
+class VisualSettingsDialog(SettingsDialog):
+    """ A dialog for visual settings manipulation, that can be uses along
```
Comment (janezd) [inline orangewidget/utils/visual_settings_dlg.py:116]: uses -> used

File: orangewidget/widget.py [hunk 2]
```diff
@@ -611,16 +609,11 @@ def _():
...
-                dlg = VisualSettingsDialog(self)
-                dlg.initialize(self.initial_visual_settings)
-                dlg.setting_changed.connect(self.set_visual_settings)
-
                 icon = QIcon(gui.resource_filename("icons/visual-settings.svg"))
                 icon.addFile(gui.resource_filename("icons/visual-settings-hover.svg"),
                              mode=QIcon.Active)
                 b = SimpleButton(icon=icon, toolTip="Set visual settings")
-                b.clicked.connect(dlg.show_dlg)
+                b.clicked.connect(self.openVisualSettingsClicked)
```
Comment (janezd) [inline orangewidget/widget.py:616]: I support the change. I feared that enforcing a certain form of dialog is to rigid, but I forgot to write so.

In practice, the signal `openVisualSettingsClicked` will most often connect to a method that will do exactly the same as the lines you removed. Perhaps you could put them into some kind of utility function? I guess the function that contains the above lines could be a static function of `SettingsDialog`, something like

```python
@staticmethod
def attach(widget):
    assert widget.initial_visual_settings is not None
    dlg = SettingsDialog(widget, widget.initial_visual_settings)
    dlg.settings_changed.connect(widget.set_visual_settings)
    widget.openVisualSettings.clicked.connect(dlg.show_dlg)
```

I obviously haven't tested this. And I may have misread/misunderstood everything.

Comment (VesnaT) [reply orangewidget/widget.py:616]: Have you seen the __init__ in VisualSettingsDialog? It does what you proposed in the `attach`.

**Pull request merged**